tests/sles4sap/ase_test.pm

```diff
@@ -0,0 +1,99 @@
+    # using all 4 conditions to determine ASE is up
+    return ($t =~ /dataserver / && $t =~ /SYBASE\.properties/ && $t =~ /SYBASE\.cfg/ && $t =~ /\-sSYBASE[^_]/);
+}
+
+sub run {
+    my ($self) = @_;
+    my $response_file = get_required_var('ASSET_1');
+    $response_file =~ s/.gz$//;
+    $self->ASE_RESPONSE_FILE($response_file);    # This module verifies SAP-ASE. We need to supply it with the response file name
+
+    # Define some variables with the scripts and files to be used while testing
+    my $showserver = '$SYBASE/$SYBASE_ASE/install/showserver';
+    my $cockpit_script = '$SYBASE/COCKPIT-4/bin/cockpit.sh';
+    my $second_log = '$HOME/second_ase_start.log';
+
+    select_serial_terminal;
+    enter_cmd 'cd';    # Let's start in $HOME
+    $self->load_ase_env;
+
+    # Is ASE running?
+    my $out = script_output $showserver;
+    die 'SAP ASE is not running' unless is_ase_up($out);
+    record_info 'showserver', $out;
+
+    # Can we connect to ASE?
+    $out = run_ase_query(query => 'select @@version');
+    die "Failed to run query. Error: [$out]" unless ($out =~ /row affected/);
+    record_info 'ASE Version', $out;
+
+    # Can ASE be stopped and started?
+    $out = run_ase_query(query => 'shutdown', timeout => 300);
+    die 'Problems detected while attempting a shutdown of ASE' unless ($out =~ /Server SHUTDOWN by request/ && $out =~ /ASE is terminating this process/);
+    $out = script_output $showserver;
+    die 'SAP ASE is still running' if is_ase_up($out);
+    # Start SAP ASE and send the output to a log file to upload to the test results
+    enter_cmd '$SYBASE/$SYBASE_ASE/install/RUN_SYBASE > ' . $second_log . ' 2>&1 &';    # Sadly, ASE start script is not daemonized, so run it with &
+                                                                                        # Wait until ASE finishes starting
+    send_key 'ret';    # Type enter to get a shell prompt
+    assert_script_run "until (grep -q 'This software contains confidential and trade secret information of SAP AG' $second_log); do sleep 5; done", timeout => 300;
+    $out = script_output $showserver;
+    die 'SAP ASE is not running' unless is_ase_up($out);
+    upload_logs $second_log;
+
+    # Check ASE cockpit
+    $out = script_output "$cockpit_script -status";
+    die 'Cockpit server is not running' unless ($out =~ /Cockpit server is running/);
+    record_info 'Cockpit Server', script_output "$cockpit_script --info";
+    assert_script_run "$cockpit_script -stop";
+    $out = script_output "$cockpit_script -status";
+    die 'Cockpit server is not stopped' unless ($out =~ /Cockpit server is stopped/);
+    # Now to start the cockpit server, it's not possible to do so and detach from the script
+    # without killing the cockpit server, so we start this in a different terminal to avoid
+    # blocking the serial terminal
+    select_console 'root-console';
```

Comment:
Are you sure we cannot have a second console and still use wait_serial?

Reply:
I'm sure that if I start the cockpit server in the serial terminal, it will block the serial terminal for further commands (at least until a Ctrl-C is sent to the serial terminal). `wait_serial` is not impacted, just the shell session opened in the serial terminal.